docs/design/rubygems.md

```diff
@@ -0,0 +1,528 @@
+# Design document for RubyGems/Bundler package manager
+
+Contents:
+
+1. [Ruby ecosystem overview](#i-ruby-ecosystem-overview)
+2. [Current implementation overview (Cachito)](#ii-overview-of-the-current-implementation-in-cachito)
+3. [Design for the Cachi2 implementation](#iii-design-for-the-implementation-in-cachi2)
+
+## I. Ruby ecosystem overview
+
+### Development prerequisites
+In order to execute the commands in the examples below, make sure you have the following packages installed in your
+environment:
+
+```bash
+sudo dnf install rubygems rubygems-bundler
+```
+
+Or use the official Ruby image from Docker hub:
+```bash
+podman run --rm -it docker.io/library/ruby:3.3.3 bash
+```
+
+### Project structure
+```bash
+bundle init # creates Gemfile in the current directory
+bundle lock # creates Gemfile.lock in the current directory
+```
+
+```bash
+├── .bundle
+│   └── config
+├── Gemfile
+├── Gemfile.lock
+├── vendor/cache
+```
+
+### Glossary
+- **Gemfile**: A file that specifies the gems that your project depends on and their versions. Bundler uses this file
+to install the correct versions of gems for your project.
+
+  ```ruby
+  source "https://rubygems.org"
+
+  gem "rails", "= 6.1.7"
+  ```
+
+- **Gemfile.lock**: A file that locks the versions of gems that are installed for your project. Bundler uses this file
+to ensure that the correct versions of gems are installed consistently across different environments.
+
+- **RubyGems**: General package manager for Ruby. Manages installation, updating, and removal of gems globally on your
```

Comment:
Or is it "RubyGem" (IDK)?

```suggestion
- **RubyGem**: General package manager for Ruby. Manages installation, updating, and removal of gems globally on your
```

Reply:
RubyGems
https://github.com/rubygems/rubygems